### DISCUSSION TITLE
Compilation Error six-library 3.1.2

### DISCUSSION THREAD

**Discussion by jwsv61099** [ngageoint/six-library#548] [General]
Hi everyone,

We are integrating our source-code into a much larger project and we are getting an error with SIX when compiling. 

For context, we are compiling using **icpc** and our source-code is compiled using g++, converted into a library and then statically linked against. 

The error is: **"/gsl/multi_span(345): error: the template argument list of the partial specialization includes a nontype argument whose type depends on a template parameter"**

It is referencing the line with the deprecated struct, **BoundsRanges**. We are using SIX Version 3.1.2. I noticed that these files were completely deprecated in the newest version, 3.1.9. Would it be best to try out the newest version? Otherwise, any ideas on how to get past this?

Thanks!

**Comment by JDanielSmith** [comment]
Yes, you should use the latest version of SIX.
